### ISSUE TITLE
Extension recommendations as modal dialogs

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#117051]
We plan to experiment with showing extension recommendations as modal dialogs after the user explicitly hit the "Set Language Mode" action.
This way user will discover extensions easier, and hopefully we will show it in just the right time.

We plan to do it start of march.

Implementation ideas:
`fileBasedRecommendations` listen on `commandService.onDidExecuteCommand` and if you see that the `workbench.action.editor.changeLanguageMode` is exectued and that the mode of the active editor is no longer `plaintext` you show the model dialog, not the notification

fyi @stevencl

**Comment by isidorn** [comment]
@jrieken had a good idea that we could also use the lightbulb which we plan to introduce in the editor title menu
@misolori can you point me to the light bulb issue? And do we have some plan when we would like to add that?

**Comment by misolori** [comment]
@isidorn that was from the ux call last week based on @Tyriar's workflow for actions-based recommendations, i'm not sure whether we had planned to add it or if it was just an exploration.

**Comment by isidorn** [comment]
I would not do anything regarding this one for this milestone.
There were some other good ideas and for now I would not yet add a modal dialog recommenation

**Comment by jrieken** [comment]
Yeah, please don't use model notifications... For me, notifications are often used for the lack of alternatives. I believe extension recommendations are important enough to deserve a dedicated UI, like contextual hints inside editors or on the activity bar icon.

**Comment by isidorn** [comment]
Yup, I agree with that.

**Comment by stevencl** [comment]
@misolori and @digitarald had a few explorations showing contextual hints in various locations. Would be good to be able to start exploring some of those ideas.

**Comment by isidorn** [comment]
I missed those. Would be cool if @misolori and @digitarald can fill me in on the details :)

**Comment by jrieken** [comment]
In some places we already have embedded hints to install extensions. Those might be good candidates to wire in recommendations. For instance, when pressing the "format document" keybinding without having a formatter we show a notification to install one. That could be enriched with recommendations and wouldn't be annoying.

**Comment by digitarald** [comment]
> showing contextual hints in various locations.

[These mockups](https://www.figma.com/proto/f6F8pniyOSRuP12MHqpjFn/Recommendations-in-Activity-Bar-Hints?node-id=1%3A1656&viewport=310%2C296%2C0.24268104135990143&scaling=min-zoom) are probably the ones @stevencl meant.

On the general idea of exposing important extensions in the right places: Format is a great example. A critical point that we we saw in usertesting was run & debug; which provides JS-related defaults no matter what file you have open. We saw users trying Chrome-debugging on a python file because they missed to install the extension before.

Many of the "important" extension recommendation enable debugging. So if the current file or workspace has recommended extension that provide debugging; users should be pointed to them during the run & debug setup (in the viewlet and the prompts). I filed #119110 to not further derail the idea of modal recommendations.
